helpers/github.ts

```diff
@@ -347,18 +362,232 @@ export async function writeTotalRewardsToGithub(statistics: Statistics) {
     }
 
     // Update or create the file
-    await octokit.rest.repos.createOrUpdateFileContents({
-      owner,
-      repo,
-      path: filePath,
-      message: "Update total rewards",
-      content: Buffer.from(content).toString("base64"),
-      sha, // Pass the SHA if the file exists, to update it
-    });
+    await octokit.rest.repos
+      .createOrUpdateFileContents({
+        owner,
+        repo,
+        path: filePath,
+        message: "Update total rewards",
+        content: Buffer.from(content).toString("base64"),
+        sha, // Pass the SHA if the file exists, to update it
+      })
+      .catch((error) => {
+        console.error(`Error updating total rewards: ${error}`);
+      });
 
     console.log(`Total rewards written to ${filePath}`);
   } catch (error) {
     console.error(`Error writing total rewards to github file: ${error}`);
     throw error;
   }
 }
+
+export async function createDevPoolIssue(projectIssue: GitHubIssue, projectUrl: string, body: string, twitterMap: TwitterMap) {
+  // if issue is "closed" then skip it, no need to copy/paste already "closed" issues
+  if (projectIssue.state == "closed") return;
+
+  // if the project issue is assigned to someone, then skip it
+  if (projectIssue.assignee) return;
+
+  // if issue doesn't have the "Price" label then skip it, no need to pollute repo with draft issues
+  if (!(projectIssue.labels as GitHubLabel[]).some((label) => label.name.includes(LABELS.PRICE))) return;
+
+  // create a new issue
+  try {
+    const createdIssue = await octokit.rest.issues.create({
+      owner: DEVPOOL_OWNER_NAME,
+      repo: DEVPOOL_REPO_NAME,
+      title: projectIssue.title,
+      body,
+      labels: getDevpoolIssueLabels(projectIssue, projectUrl),
+    });
+    console.log(`Created: ${createdIssue.data.html_url} (${projectIssue.html_url})`);
+
+    if (!createdIssue) {
+      console.log("No new issue to tweet about");
+      return;
+    }
+
+    // post to social media
+    try {
+      const socialMediaText = getSocialMediaText(createdIssue.data);
+      const tweetId = await twitter.postTweet(socialMediaText);
+
+      twitterMap[createdIssue.data.node_id] = tweetId?.id ?? "";
+      await writeFile("./twitterMap.json", JSON.stringify(twitterMap));
+    } catch (err) {
+      console.error("Failed to post tweet: ", err);
+    }
+  } catch (err) {
+    console.error("Failed to create new issue: ", err);
+    return;
+  }
+}
+
+export async function handleDevPoolIssue(
```

Comment:
This function truly is massive and would benefit to be broken down into smaller parts